org.eclipse.xtext.xbase.ide/src/org/eclipse/xtext/xbase/ide/contentassist/XbaseIdeCrossrefProposalProvider.xtend

```diff
@@ -44,7 +44,7 @@ class XbaseIdeCrossrefProposalProvider extends IdeCrossrefProposalProvider {
 					0
 			val converter = getTypeConverter(context.resource)
 			val objectOrProxy = candidate.getEObjectOrProxy
-			return proposalCreator.createProposal(proposalString, context) [ result |
+			val proposal = proposalCreator.createProposal(proposalString, context) [ result |
```

Comment:
Any reason not to move this to the closure?

Reply:
Am not sure if the create proposal can return null if prefix is bad

Reply:
My fault, didn't notice that this was an initialisation closure. If I do it in the closure, it's nullsafe.
Will replace my current implementation for the one you proposed.